spec/datadog/core/telemetry/v1/app_started_spec.rb

```diff
@@ -122,4 +122,39 @@
       end
     end
   end
+
+  describe '#to_h' do
+    subject(:to_h) { app_started.to_h }
+
+    context 'when attributes are all nil' do
+      let(:additional_payload) { nil }
+      let(:configuration) { nil }
+      let(:dependencies) { nil }
+      let(:integrations) { nil }
+      it { is_expected.to eq({}) }
+    end
+
+    context 'when attributes are all defined' do
+      let(:additional_payload) { { 'tracing.enabled' => true, 'profiling.enabled' => false } }
+      let(:configuration) { { DD_AGENT_HOST: 'localhost', DD_TRACE_SAMPLE_RATE: '1' } }
```

Comment:
These inputs don't seem very... realistic? E.g. according to the documentation in `#initialize` and other tests, `additional_payload` and `configuration` are expected to be both arrays (here we're just using hashes) and instances of `Telemetry::V1::Configuration`.

I actually think we may be hiding a bug here with these tests, since we're using `map_hash` in the code for these two fields and it seems like it should be `map_array`?

Reply:
Based on https://github.com/DataDog/dd-trace-rb/pull/2102#discussion_r906509004, it was decided to change `additional_payload` and `configuration` to simple hashes. This change is reflected in `collector.rb` but not anywhere else. I was going to modify the rest of the code to reflect this in a separate PR, but I can make those changes in this one instead!

Reply:
Ah, got it. I think it's better to do it an separate PR, since this one is reasonably scoped and in a good shape :)